relay-filter/src/config.rs

```diff
@@ -328,15 +482,208 @@ impl FiltersConfig {
+    /// isolation, they don't provide enough valuable information to perform the
+    /// filtering. Additionally, new versions may include features not supported
+    /// in the current Relay.
+    fn is_version_supported(&self) -> bool {
+        if self.config.max_version < self.config.project_filters.version {
+            return false;
+        }
+        if let Some(gc) = self.config.global_filters {
+            if self.config.max_version < gc.version {
+                return false;
+            }
+        }
+        true
+    }
+
+    fn next_project_filter(&mut self) -> Option<&'a GenericFilterConfig> {
+        let Some((id, filter)) = self
+            .config
+            .project_filters
+            .filters
+            .get_index(self.project_index)
+        else {
+            return None;
+        };
+        self.project_index += 1;
+
+        // Mark the filter as evaluated first. If the filter is disabled, it
+        // should not be applied from global configs.
+        let newly_evaluated = self.evaluated.insert(id);
+
+        if !filter.is_enabled || !newly_evaluated {
+            return None;
+        }
+
+        if filter.condition.is_none() {
+            let Some(global_filters) = self.config.global_filters else {
+                return None;
+            };
+            let Some(filter) = global_filters.filters.get(id) else {
+                return None;
+            };
+            return Some(filter);
+        }
+
+        Some(filter)
+    }
+
+    fn next_global_filter(&mut self) -> Option<&'a GenericFilterConfig> {
+        let Some(global_filters) = self.config.global_filters else {
+            return None;
+        };
+        let Some((id, filter)) = global_filters.filters.get_index(self.global_index) else {
+            return None;
+        };
+        self.global_index += 1;
+
+        if filter.is_empty() {
+            return None;
+        }
+
+        let newly_added = self.evaluated.insert(id);
+        if !newly_added {
+            return None;
+        }
+
+        Some(filter)
+    }
+}
+
+impl<'a> Iterator for CombinedFiltersConfigIter<'a> {
+    type Item = &'a GenericFilterConfig;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if !self.is_version_supported() {
+            return None;
+        }
+
+        loop {
+            if self.project_index < self.config.project_filters.filters.len() {
+                let filter = self.next_project_filter();
+                if filter.is_some() {
+                    return filter;
+                }
+                continue;
+            }
+
+            if let Some(global_filters) = self.config.global_filters {
+                if self.global_index < global_filters.filters.len() {
+                    let filter = self.next_global_filter();
+                    if filter.is_some() {
+                        return filter;
+                    }
+                    continue;
+                }
+            }
```

Comment:
Does this mean we have one filter from project config and then from global config, and next iteration the same? 
Don't we want to iterate first all project config filters and only then all global config filters? 

Or maybe I"m getting this wrong?